lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -2109,6 +2116,41 @@ sub done ($self, %args) {
     return $new_val{result} // $self->result;
 }
 
+sub _reason_and_result ($self, $new_val, $result, $reason, $restart) {
+    my $result_unknown = $self->result eq NONE;
+    my $reason_unknown = !$self->reason;
+    $new_val->{result} = $result if $result_unknown;
+    if (($result_unknown || $reason_unknown) && defined $reason) {
+        # restart incompletes when the reason matches the configured regex
+        my $append_reason = '';
+        my $auto_clone_regex = OpenQA::App->singleton->config->{global}->{auto_clone_regex};
+        if ($result eq INCOMPLETE and $auto_clone_regex and $reason =~ $auto_clone_regex) {
+            my $limit = OpenQA::App->singleton->config->{global}{auto_clone_limit};
+            my $ancestors = $self->incomplete_ancestors($limit + 1);
+            # how many of those incomplete ancestors had a reason not matching auto_clone?
+            my $unrelated = grep { $_->[2] !~ m/$auto_clone_regex/ } @$ancestors;
+            if (@$ancestors < $limit || $unrelated > 0) {
+                $append_reason = ' [Auto-restarting because reason matches auto_clone_regex]';
+                $$restart = 1;
+            }
+            else {
+                $append_reason
+                  = ' [Not restarting because job has been restarted already auto_clone_limit times and failed with /$auto_clone_regex/]';
```

Comment:
```suggestion
                  = ' [Not restarting because job has been restarted already $auto_clone_limit times and failed with /$auto_clone_regex/]';
```

as discussed